M66/M66ATParser/M66ATParser.cpp

```diff
@@ -258,7 +258,9 @@ bool M66ATParser::close(int id)
     //May take a second try if device is busy
     for (unsigned i = 0; i < 2; i++) {
         if (tx("AT+QICLOSE=%d", id)
-            && scan("%d, CLOSE OK", &id_resp)) {
+            && scan("%d, CLOSE OK", &id_resp)
+            && (id == id_resp)) {
+
             return true;
```

Comment:
Do the same here as in the other function, return id==id_resp instead of checking in the if clause. It also looks to me as if we return an undefined value if close didn't work. The last line in this function should be return false;